### PR TITLE
Made the changes per issue #217

### LINKED ISSUE #217: Add ability to edit point attributes on a line chart.
### Add ability to edit Point attributes on the line graphs
First off, I love the charting in PSHTML! I'd also like if it was possible to edit the point attributes on a line chart.

Specifically, looking at the [Chart.js docs](https://www.chartjs.org/docs/latest/charts/line.html) they mention several attributes that I'd like to be able to set with PSHTML:

```text
pointRadius (this is hard coded to 4 in Class datasetline)
pointHitRadius
pointHoverRadius
```

### Steps to reproduce
Create a line chart and the script will contain:

```javascript
"pointRadius":4, ... "pointHitRadius":null, ... "pointHoverRadius":0
```

I'm happy to take this issue, but wanted to submit it before working on it.

### PR DESCRIPTION
**Pull request by ThePoShWolf**
# Pull Request [Topic]

Added the ability to set the point settings for a line chart.

I also updated the line chart docs to include a reference to the new parameters and I fixed the example script so that it works.

### Please tell us , the type of Change you are submitting:
Select one of the following: 

- [ ] Bug
- [ ] Feature
- [X] Minor Change

### Does it fix an existing issue? Please tell us which one

Fixes #217 

### Description of what's been changed
You can now set the following attributes on a line chart:
- PointRadius
- PointHitRadius
- PointHoverRadius

This was done by adding SetPointSettings() and GetPointSettings() to the DataSetLine class and adding the appropriate parameters to New-PSHTMLChartLineDataset.

### Results of your tests (If applicable)
Passed:
image

### CODE REVIEW

File: Code/Classes/003.pshtml.components.charts.ps1 [hunk 1]
```diff
@@ -125,17 +125,17 @@ Class datasetline : dataset{
...
     $pointBorderColor = "rgb(0,0,0)"
     [Int[]]$pointBorderWidth = 1
-    [Int]$pointRadius = 4
+    [float]$pointRadius = 4
     [ValidateSet("circle","cross","crossRot","dash","line","rect","rectRounded","rectRot","star","triangle")]
     $pointStyle = "circle"
 
     [int[]]$pointRotation
-    [int[]]$pointHitRadius
+    [float]$pointHitRadius
```
Comment (Stephanevg) [inline Code/Classes/003.pshtml.components.charts.ps1]: [float]$pointHitRadius = $null

**Comment by Stephanevg** [comment]
Hi @ThePoShWolf thank you SO much for this PR.

Sorry I am so late in adressing this one, but, you know, life. huh ;)


I do have some build errors. One is because the output of the JSON object is different as prior your modifications.
I actually think that this is an easy fix, and would be great you can fix it directly too.

# Wrong default value (?)

See here for the error detail:

https://ci.appveyor.com/project/Stephanevg/pshtml/builds/23206855/job/lpwc4ns6ay93rqxx/tests

As  you can see on this screen shot, the default value of PointHitRadius is not `$null` anymore but `0`. 

image

I think that is because of the fact that you changed the type to `[float]` and it now has a default value '0' instead of null. You could potentially change either the script, or the test. As you wish.

# Branch

Could you please also push the PR to the `dev` branch instead of the master one ?
I do have some build failures on the Linux apveyor machine since their last update, and I don't want to miss this pr.

Cheers

**Comment by ThePoShWolf** [comment]
@Stephanevg Alright, I did some testing to be sure that a default value of 0 wouldn't mess things up, and as far as I can tell, that should be ok. So I switched the test to check for a default value of 0:

image

Also switched it to the dev branch, my bad 😄

**Comment by Stephanevg** [comment]
Hi @ThePoShWolf  I know this is a SUPER old PR. But I was planning to merge it soon for version 0.8.0. Do you mind doing a pul, and resolving the conflicts before I merge? I would much appreciate that! :)

**Comment by ThePoShWolf** [comment]
@Stephanevg Definitely! My apologies for not fixing this sooner! I'll have it done this week.

**Comment by ThePoShWolf** [comment]
Hmm... Got the conflicts resolved, but now the tests are failing:

```powershell
Install-NuGetClientBinaries : Exception calling "ShouldContinue" with "2" 
argument(s): "Windows PowerShell is in NonInteractive mode. Read and Prompt 
functionality is not available."
At C:\Program 
Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:1322 char:9
+         Install-NuGetClientBinaries -CallerPSCmdlet $PSCmdlet
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-NuGetClientBinaries], 
    MethodInvocationException
    + FullyQualifiedErrorId : PSInvalidOperationException,Install-NuGetClientB 
   inaries
```

After a little investigation, turns out I botched the resolution of the merge conflicts. I believe that is fixed now 😄

**Comment by ThePoShWolf** [comment]
Ok, now the tests are actually failing:

```powershell
[-] [New-PSHTMLChart][-Type Line][-DataSet LineDataSet][Label][Title][CanvasId] Should create ChartJS javascript Code 54ms
      Expected strings to be the same, but they were different.
      Expected length: 1000
      Actual length:   1003
      Strings differ at index 561.
      Expected: '...ius":0,"Po...'
      But was:  '...ius":null,...'
      504:         $Is | should be $Should
```

Let me get that fixed.

**Comment by ThePoShWolf** [comment]
image
Ok, works locally with the latest changes, lets see how it works in the pipeline.

**Comment by ThePoShWolf** [comment]
Bah, I missed other fails before:

```powershell
[-] [DatasetLine][Methods][SetPointSettings][Float] Should not throw 56ms
      Expected no exception to be thrown, but an exception "Method invocation failed because [datasetline] does not contain a method named 'SetPointSettings'." was thrown from D:\a\1\s\Tests\Chart.DatasetLine.Tests.ps1:59 char:10
          +         {$d.SetPointSettings(4,0,0)} | should not throw
          +          ~~~~~~~~~~~~~~~~~~~~~~~~~~.
      59:         {$d.SetPointSettings(4,0,0)} | should not throw
      at <ScriptBlock>, D:\a\1\s\Tests\Chart.DatasetLine.Tests.ps1: line 59
    [-] [DatasetLine][Methods][SetPointSettings][Float] Should Set Point Settings 7ms
      RuntimeException: Method invocation failed because [datasetline] does not contain a method named 'SetPointSettings'.
      at <ScriptBlock>, D:\a\1\s\Tests\Chart.DatasetLine.Tests.ps1: line 64
```

Investigating.

**Comment by ThePoShWolf** [comment]
Interesting. Windows PS and PS Core are outputting the float as a string differently, which is causing the tests to fail in Windows PS, even though they work in PS core.

Windows PS: '4'
PS Core: '4.0'

I'll have to take a look at this again when I have some more time.

**Comment by Stephanevg** [comment]
Hi @ThePoShWolf 

I looked into this quickly: here a screen shot of the failing test (just for reference)

image



Do you think this is a bug on Microsoft's side?

Does this has an influence on the feature? (like can ChartJS not render the chart correct when there is only 4 instead of 4.0 ?)

If not, do you think we just a conditional test on the powershell version for this particular test?

Cheers

**Comment by ThePoShWolf** [comment]
Hey @Stephanevg I was looking at that test as well, I believe the easiest work around is going to be to have that particular test check for the PS version and use '4.0' for PS 6+ and '4' for PS 5 and below. Do you think that is reasonable?

Also for ChartJS, I did some testing and it will accept decimals or whole numbers and draw accordingly.

**Comment by Stephanevg** [comment]
Hi @ThePoShWolf Yeah, that would be completly fine actually. Go ahead! :)

**Comment by Stephanevg** [comment]
Hi @ThePoShWolf I have done some few modifications on the main code in the psm1. Would be great if you can pull them and fix the conflicts one last time.
I'll merge it immediatley, although that one last test is not passing yet.
I would like to get a preview out this weekend, and that PR is just the one last thing I need to merge :)

**Pull request merged**